Spec/Auth.swift

```diff
@@ -2160,18 +2163,19 @@ class Auth : QuickSpec {
                     expect(params.timestamp).to(equal(NSDate(timeIntervalSince1970: 123)))
                 }
 
-                it("if not explicitly set, should be generated at the getter and stick") {
+                it("if not explicitly set, should be generated at the initializer and stick") {
                     let params = ARTTokenParams()
 
                     waitUntil(timeout: testTimeout) { done in
+                        let now = NSDate().artToIntegerMs()
+                        guard let timestamp = params.timestamp else {
```

Comment:
I am not sure I follow this.  If a user creates a TokenParams object, if I am reading this correctly, you are setting a timestamp as part of the token params.  So if the user then did something async, and later did an auth using the TokenParams, the timestamp woudl be the timestamp the TokenParams object was created? If so, that is wrong.  The vanilla TokenParams object has no timestamp value unless explicitly set. Then when a token request is issued, if the TokenParams object has a value it uses it, if not the current timestamp is used. There is no benefit in issuing a timestamp when the TokenParams object is initialized whatsoever, it will simply mean that it's more likely that by the time a token request is created, the timestamp will be out of date.